tests/saw/spec/HMAC_append_correct.saw

```diff
@@ -30,14 +30,44 @@ let hash_lemma2 =
 
 let mod_lemma = rewrite (cryptol_ss ()) (unfold_term ["modthm"] {{ modthm }});
 
+let tuple_lemma = rewrite (cryptol_ss ()) (unfold_term ["modthm"] {{ modthm }});
```

Comment:
What is the point of these lemmas?

Reply:
These are part of the proof that the init/update/final structure of s2n's HMAC matches the high-level, single-function spec. To do this, we have a high-level one-function spec in Cryptol and a lower-level three-function spec in Cryptol, along with a proof that executing init -> update* -> final for any decomposition of a message is equivalent to the one-function call on the entire message. We do that proof inside the rewriting system in SAW, which is somewhat impoverished, so we sometimes have to create somewhat obvious intermediate lemmas to allow the proof to go through. When the structure of terms changes at all (as it did in this case), sometimes the necessary lemmas change.